compiler/frontend/src/module/module.rs

```diff
@@ -10,29 +10,59 @@ use std::{
     fs,
     hash::Hash,
     path::{Path, PathBuf},
+    sync::Arc,
 };
 use tracing::{error, warn};
 
 #[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
-pub struct Module {
-    pub package: Package,
-    pub path: Vec<String>,
-    pub kind: ModuleKind,
+pub struct Module(Arc<InnerModule>);
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+struct InnerModule {
```

Comment:
The convention seems to be to use `Inner` as a suffix

```suggestion
struct ModuleInner {
```